app/src/common/shared/com/igalia/wolvic/utils/BitmapCache.java

```diff
@@ -32,6 +32,14 @@ public class BitmapCache {
     private Surface mCaptureSurface;
     private boolean mCapturedAcquired;
 
+    private String normalizeKey(@NonNull String input) {
+        // Refer to https://stackoverflow.com/a/33994701
+        final int MAX_LENGTH = 64;
+         return (input.length() > MAX_LENGTH ? input.substring(0, MAX_LENGTH) : input)
```

Comment:
Wouldn't this cause collisions in some cases? For example, with URLs like this:

```
https://www.example.com/this/is/a/long/path/to/the/image/folder/picture_1.jpg
https://www.example.com/this/is/a/long/path/to/the/image/folder/picture_2.jpg
```

Maybe we could use a checksum if the input is too long:

```java
private String normalizeKey(@NonNull String input) {
    final int MAX_LENGTH = 64;
    if (input.length() > MAX_LENGTH) {
        // To prevent collisions, use a checksum if the input is too long.
        CRC32 crc = new CRC32();
        crc.update(input.getBytes());
        return Long.toHexString(crc.getValue());
    } else {
        // Refer to https://stackoverflow.com/a/33994701
        return input
            .toLowerCase()                  // All alphabetic ascii to lowercase
            .replaceAll("[^a-z0-9_-]","_"); // Replace not permitted char with "_"
    }
}
```

Reply:
Updated, thank you for suggesting it!